core/server/services/auth/api-key/admin.js

```diff
@@ -0,0 +1,82 @@
+const jwt = require('jsonwebtoken');
+const models = require('../../../models');
+const {
+    InternalServerError,
+    UnauthorizedError
+} = require('../../../lib/common/errors');
+
+const JWT_OPTIONS = {
+    maxAge: '5m',
+    algorithms: ['HS256']
+};
+
+/**
+ * Remove 'Bearer' from raw authorization header and extract the API Key ID and
+ * JWT token. Eg. Authorization: Bearer ${objectId}:${JWT}
+ * @param {string} header
+ */
+const extractCredentialsFromHeader = function extractCredentialsFromHeader(header) {
+    let [scheme, credentials] = header.split(' ');
+
+    if (/^Bearer$/i.test(scheme)) {
+        let [apiKeyId, token] = credentials.split('|');
+
+        return {
+            apiKeyId,
+            token
+        };
+    } else {
+        return {};
+    }
+};
+
+const authenticateAdminAPIKey = function authenticateAdminAPIKey(req, res, next) {
+    // allow fallthrough to other auth methods or final ensureAuthenticated check
+    if (!req.headers || (req.headers && !req.headers.authorization)) {
+        return next();
+    }
+
+    let {apiKeyId, token} = extractCredentialsFromHeader(req.headers.authorization);
+
+    if (!apiKeyId || !token) {
+        return next(new UnauthorizedError({
+            message: 'Authorization header format is "Authorization: Bearer [api key id]|[token]"',
+            code: 'INVALID_AUTH_HEADER'
+        }));
+    }
+
+    models.ApiKey.findOne({id: apiKeyId}).then((apiKey) => {
+        if (!apiKey) {
+            return next(new UnauthorizedError({
+                message: 'Unknown API Key',
+                code: 'UNKNOWN_API_KEY'
+            }));
+        }
+
+        // TODO: should we do Buffer.from(x, 'hex') or is using the secret as-is ok?
```

Comment:
I think we probably should? Although `jsonwebtoken` is fine using a plain string according to their [readme comment](https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback) it does suggest that other libraries need random chars as mentioned in the [linked comment](https://github.com/auth0/node-jsonwebtoken/issues/208#issuecomment-231861138)